### ISSUE TITLE
Extension does not install dotnet

### ISSUE THREAD

**Issue by mikaelsand** [Azure/LogicAppsUX#3880]
### Describe the Bug with repro steps

1. Have VS Code
2. Have dotnet installed on machine.
3. Remeber that the Logic Apps Standard extension was updated.
4. Uninstall the extension.
5. Run the new extension accroding to [the official instructions](https://learn.microsoft.com/en-us/azure/logic-apps/create-single-tenant-workflows-visual-studio-code#tools).
6. When starting the extension I get this message:
```ps
NodeJs Binaries: C:\Users\u-mik-0001\.azurelogicapps\dependencies\NodeJs
FuncCoreTools Binaries: C:\Users\u-mik-0001\.azurelogicapps\dependencies\FuncCoreTools
2:40:19 PM: Running command: "C:\Users\u-mik-0001\.azurelogicapps\dependencies\DotNetSDK\dotnet.exe --version"...
DotNetSDK Binaries: C:\Users\u-mik-0001\.azurelogicapps\dependencies\DotNetSDK
'C:\Users\u-mik-0001\.azurelogicapps\dependencies\DotNetSDK\dotnet.exe' is not recognized as an internal or external command,
operable program or batch file.
2:40:19 PM: Azure Logic Apps Standard Runtime Dependencies validation and installation completed successfully.
```
Note that the script says the validation was completed successfully even if there is an error.
7. Look in the dependencies folder: C:\Users\username\.azurelogicapps\dependencies\DotNetSDK. It is empty.
8. Dotnet is still installed. The command `dotnet --list-sdks` gives this list:
```text
3.1.426 [C:\Program Files\dotnet\sdk]
6.0.417 [C:\Program Files\dotnet\sdk]
7.0.307 [C:\Program Files\dotnet\sdk]
7.0.400 [C:\Program Files\dotnet\sdk]
8.0.100 [C:\Program Files\dotnet\sdk]
```
9. Uninstall and reinstall the extension, does not solve it.
10. Install the extension using the vix-file, does not solve it.

### What type of Logic App Is this happening in?

Standard (VSCode)

### Are you using new designer or old designer

New Designer

### Did you refer to the TSG before filing this issue? https://aka.ms/lauxtsg

Yes

### Browser

No browser

**Comment by ccastrotrejo** [comment]
Hi @mikaelsand thanks for raising this issue, I am wondering if something happened while the installation of dotnet that prevented to download and unzipping of the binaries into the dependencies path. In the meantime, as a workaround, could you please do the following:

1. Delete the whole `dependencies` folder. 
2. Relaunch/close and open vscode

When you delete the `dependencies` folder and relaunch vscode, out extension will try to install again all of the dependencies (nodejs, func core tools and dotnet) 

Please let me know if this doesn't work

**Comment by mikaelsand** [comment]
So close!
I deleted the dependencies and restarted VS Code and everything started to download and it worked fine until ... Here is the log:

```ps
NodeJs Binaries: C:\Users\u-mik-0001\.azurelogicapps\dependencies\NodeJs
Creating temporary folder... C:\Users\U-MIK-~1\AppData\Local\Temp\.azurelogicapps\NodeJs
Downloading binaries from: https://nodejs.org/dist/v18.17.1/node-v18.17.1-win-x64.zip
Successfullly downloaded NodeJs binaries.
Extracting C:\Users\U-MIK-~1\AppData\Local\Temp\.azurelogicapps\NodeJs\NodeJs.zip
Extraction NodeJs successfully completed.
Removed C:\Users\U-MIK-~1\AppData\Local\Temp\.azurelogicapps\NodeJs
FuncCoreTools Binaries: C:\Users\u-mik-0001\.azurelogicapps\dependencies\FuncCoreTools
Creating temporary folder... C:\Users\U-MIK-~1\AppData\Local\Temp\.azurelogicapps\FuncCoreTools
Downloading binaries from: https://github.com/Azure/azure-functions-core-tools/releases/download/4.0.5198/Azure.Functions.Cli.win-x64.4.0.5198.zip
9:45:56 AM: Error in dependencies validation and installation: "{0}"...
```

So I tried to start a new project and got this error:

```ps
9:46:48 AM: Running command: "C:\Users\u-mik-0001\.azurelogicapps\dependencies\DotNetSDK\dotnet.exe --version"...
The system cannot find the path specified.
9:46:48 AM: Error: You must have the [.NET Core SDK](https://aka.ms/AA4ac70) installed to perform this operation. See [here](https://aka.ms/AA1tpij) for supported versions.
```

The link points to dotnet 8 by the way.
I already have all the SDKs installed. See earlier post.

Can you please advice further?

**Comment by FlipFlop17** [comment]
The same thing happened to me when i first installed the extension a month ago. Dotnet, Function app, nodejs folder were just empty.
What I did is installed all of the dependencies manually and copied them inside the .dependencies folder accordingly and that solved it for me.

**Comment by mikaelsand** [comment]
@FlipFlop17 That is great! Where do I copy the dependency from? It is looking for dotnet.exe but that file is not in the specific dotnet version folder. It is the dotnet root folder. Do I copy _everything_ in the standrad dotnet folder to the dependencies folder?

Example: 
Dotnet-folder: `C:\Program Files\dotnet`

Logic Apps needs 6.x SDK.Which is in `C:\Program Files\dotnet\sdk\6.0.417` but there is no`dotnet.exe` in that folder.

**Comment by FlipFlop17** [comment]
You should have:
C:\Users\%user%\.azurelogicapps\dependencies
You should see  inside the dotnet, azurefunction, nodejs folder.

If those folders are empty, like it was in my case, that could be the issue.
If the folders are not empty means the problem lies somewhere else.

Download manually :
https://github.com/Azure/azure-functions-core-tools/releases/tag/4.0.5455
https://nodejs.org/en/download

and unpack/paste it to those folders accordingly and try to open the app.

If you have dotnet.exe already installed you can just copy everything to C:\Users\%user%\.azurelogicapps\dependencies\dotnet folder. You should have dotnet.exe inside 'C:\Program Files\dotnet\sdk\6.0.417' maybe try reinstalling.

https://learn.microsoft.com/en-us/dotnet/core/install/windows?tabs=net80#install-with-windows-installer

**Comment by mikaelsand** [comment]
Naw. Reinstalling dotnet 6 did not work. I had the option to repair and the dotnet.exe file is still in the root of dotnet (as it should be is my impression).

So what I did was to copy the entire 2.2 GB of ditnet into the logic app dependencies folder. It works, however that is not the soultion I was hoping for. Can someone else advice?

**Comment by mikaelsand** [comment]
Adding some information: After I copied everything dotnet into the dependences folder, it works but when I try to open a workflow designer I get this error:

```ps
2:16:22 PM: Running command: "C:\Users\u-mik-0001\.azurelogicapps\dependencies\FuncCoreTools\func host start --port 8000" with pid: "29900"...
This program is blocked by group policy. For more information, contact your system administrator.
```

I do not know if it's related in any way.

**Comment by FlipFlop17** [comment]
you are on your work computer? when the designer starts it starts the func.exe (azure core function) in the background. it needs it to run successfully. Never seen this error, it could literally mean that the exe is blocked if you are working on your work computer and security is tight.

**Comment by mikaelsand** [comment]
@FlipFlop17 Yes I am running this on my work computer and that gave me an idea. I ran VS Code as admin and all of a sudden: The installation worked! 

+ Everything seems to be installed correctly.
- I had to run as admin for it to work.

And now I have other issues, but I will open a new issue for that.
Thank you very much for your help <3
